packages/celotool/src/lib/oracle.ts

```diff
@@ -341,45 +332,83 @@ function getOracleAzureIdentityName(keyVaultName: string, address: string) {
 
 /**
  * Fetches the env vars for a particular context
- * @param context the OracleAzureContext to use
+ * @param oracleContext the oracle context to use
  * @return an AzureClusterConfig for the context
  */
-export function getAzureClusterConfig(context: OracleAzureContext): AzureClusterConfig {
-  const configEnvVars = oracleAzureContextClusterConfigEnvVars[context]
-  const clusterConfig: AzureClusterConfig = {
-    subscriptionId: '',
-    tenantId: '',
-    resourceGroup: '',
-    clusterName: '',
-  }
-  for (const k of Object.keys(configEnvVars)) {
-    const key = k as keyof AzureClusterConfig
-    clusterConfig[key] = fetchEnv(configEnvVars[key])
-  }
-  return clusterConfig
+export function getAzureClusterConfig(oracleContext: string): AzureClusterConfig {
+  return getOracleContextDynamicEnvVarValues(oracleContextAzureClusterConfigDynamicEnvVars, oracleContext)
 }
 
 /**
- * Given if the desired context is primary, gives the appropriate OracleAzureContext
+ * Gives an object with the values of dynamic environment variables for an oracle context.
+ * @param dynamicEnvVars an object whose values correspond to the desired
+ *   dynamic env vars to fetch.
+ * @param oracleContext The oracle context
+ * @param defaultValues Optional default values if the dynamic env vars are not found
+ * @return an object with the same keys as dynamicEnvVars, but the values are
+ *   the values of the dynamic env vars for the particular oracleContext
  */
-export function getOracleAzureContext(primary: boolean): OracleAzureContext {
-  return primary ? OracleAzureContext.PRIMARY : OracleAzureContext.SECONDARY
+export function getOracleContextDynamicEnvVarValues<T>(
+  dynamicEnvVars: { [k in keyof T]: DynamicEnvVar },
+  oracleContext: string,
+  defaultValues?: { [k in keyof T]: string }
+): {
+  [k in keyof T]: string
+} {
+  return Object.keys(dynamicEnvVars).reduce(
+    (values: any, k: string) => {
+      const key = k as keyof T
+      const dynamicEnvVar = dynamicEnvVars[key]
+      const dynamicEnvVarName = getDynamicEnvVarName(dynamicEnvVar, {
+        oracleContext
+      })
+      const defaultValue = defaultValues ? defaultValues[key] : undefined
+      const value = defaultValue !== undefined ?
+        fetchEnvOrFallback(dynamicEnvVarName, defaultValue) :
+        fetchEnv(dynamicEnvVarName)
+      return {
+        ...values,
+        [key]: value,
+      }
+    },
+  {})
 }
 
 /**
  * Switches to the AKS cluster associated with the given context
  */
-export function switchToAzureContextCluster(celoEnv: string, context: OracleAzureContext) {
-  const azureClusterConfig = getAzureClusterConfig(context)
+export function switchToAzureContextCluster(celoEnv: string, oracleContext: string) {
+  const validOracleContexts = fetchEnv(envVar.ORACLE_CONTEXTS)
```

Comment:
I like having this check! I can see this being used for a switchToAwsContextCluster function. Perhaps lines 381-386 can be made into a new "isValidOracleContext" function for reusability and possibly better readability?